apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1867,9 +1868,13 @@ func (u *UniterAPI) prepareRelationResult(rel *state.Relation, applicationName s
 	for _, otherEp := range otherEndpoints {
 		otherAppName = otherEp.ApplicationName
 	}
+	modelInfo, err := u.modelInfoService.GetModelInfo(context.TODO())
```

Comment:
This `context.TODO()` can be replaced with the actual context, it's only two levels above and available on every call site.

Reply:
Also, can `modelInfo` be passed as argument? I see this method is called on a loop for example and so this means N db hits.

Reply:
We could get it further up the call stack. However that would mean getting data before confirming that the unit has authorization to use this facade call. I'm on the fence.